File: Java/DSA/Sorting/InsertionSort.java
package Sorting;

import java.util.Arrays;

public class InsertionSort {

    public static void main(String[] args) {
        int[] data = { 9, 5, 1, 4, 3 };
        insertionSort(data);
        System.out.println(Arrays.toString(data));
    }
    static void insertionSort(int[] arr){

        int  size = arr.length;

        for (int i=0;i<size;i++){
            int key = arr[i];
            int j = i-1;

            while(j >= 0 && key < arr[j]){
                arr[j+1] = arr[j];
                j--;
            }

            arr[j+1] = key;
        }

    }


}

File: Python/DSA/Sorting/insertion_sort.py
def insertion_sort(arr):
    size = len(arr)

    for i in range(1,size):
        key = arr[i]
        j = i-1
        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j-=1
        
        arr[j+1] = key

arr =  [9, 5, 1, 4, 3]
insertion_sort(arr)
print(arr)